### DISCUSSION TITLE
Build size 2.0.17 vs 3.0.1

### DISCUSSION THREAD

**Discussion by InnuendoPi** [espressif/arduino-esp32#9860] [Q&A] [answered]
Hey,
compiled code with 3.0.1 is 10% greater than same source code in a 2.0.17 env. Can't find a reason. 3.0.1 app bin files won't fit my custom partition table. My env: Arduino IDE, Arduino CLI, VSCode with Arduino extension and a modified ESP8266FS with LittleFS support. toolchain ESP-IDF not installed. 
First I thought about loglevels in libs (extra code), but i can't find differences in git history between 2.0.x and 3.0.x
Any tips how to downsize app.bin are welcome.
regards Innu

**Comment by me-no-dev** [comment, marked as answer]
Unfortunately the WiFi lib (and other IDF components) grew from IDF 4.4 to IDF 5.1. WiFi alone grew by about 200KB because of support for newer encryption methods, etc. Nothing we can do really.

**Reply by InnuendoPi** [reply]
thanks for fast response!

**Comment by daledj** [comment]
I have a similar issue. My ESP32-C3 code complied and ran with a binary size of 860 KB in 2.0.17, but in 3.0.1 its size is 1084 KB. I was able to update the smaller file OTA and now it seems that OTA fails, strangely with the error of "file not found" even though the file is on the OTA server. I've read in https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-guides/performance/size.html about ways to reduce the size of the WiFi related portions of the file, but I can find no information on how to modify the configuration in the Arduino IDE to make the optimizations discussed in the espressif documentation. Can someone point me to how to do this in the Arduino IDE?

**Reply by me-no-dev** [reply]
We have done all we can to minimize the size, without removing features that we actually need. Most options are for building custom firmwares for specific environments. We can not go specific and rather need to stay generic, so that features work for everyone. You have two ways to tackle this:
1. Generate custom libs to use in Arduino: https://docs.espressif.com/projects/arduino-esp32/en/latest/lib_builder.html
2. Use Arduino as ESP-IDF component: https://docs.espressif.com/projects/arduino-esp32/en/latest/esp-idf_component.html

**Reply by daledj** [reply]
Thank you for your quick response. I can appreciate the need to stay generic.

**Comment by cziter15** [comment]
There's some extra size added by chip debug report. Also, there's a bug that debug logs are not properly stripped out from the binary even if core debug level is set to 0.

**Reply by me-no-dev** [reply]
you mean IDF logs, right? not Arduino

**Reply by cziter15** [reply]
I mean this:

https://github.com/espressif/arduino-esp32/blob/9e55ccd98ee7f202766bbd133000abbed38a065d/cores/esp32/esp32-hal-log.h#L125

Using 3.0.1 with platformio, these strings are in binary despite core log level set to 0.

**Reply by me-no-dev** [reply]
you see it's all preprocessor, so why it's there I can not say. Have you tried in Arduino IDE/CLI?

**Reply by cziter15** [reply]
after double checking, yup, it seems to me something wrong with defines on my side, my bad :)
anyway, chip-debug-report might be a good candidate for #ifdef at specific log level

**Reply by me-no-dev** [reply]
you will not save much by that. We thought about it (it already is limited in a way by the log level), but we also want to trigger it manually for users who have issues setting the log level.

**Comment by InnuendoPi** [comment]
@me-no-dev can you please make a final release 2.0.18 with ESP-IDF 4.4.8? ESP-IDF 4.4.8 is the last update/fix (EOL). For a couple of hours you changed the topic to 2.0.18 before you closed PR 4.4
